### PR TITLE
Docs/v23 upgrade guide miscellaneous changes section

### PR DESCRIPTION
**Pull request by rolfsmeds**

### CODE REVIEW

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc [hunk 1]
```diff
@@ -134,21 +134,49 @@ A set of API breaking changes and their replacements are listed below:
...
+
+// Add dialogs explicitly to UI
+add(firstDialog, secondDialog);
+
+// Later open dialogs
+firstDialog.open();
+secondDialog.open();
+----
+
+-  Positive tabindex values (such as `setTabIndex(5)`) are no longer supported on input fields. Doing this will not cause errors, but will simply not have any effect on tab order.
```
Comment (tomivirkki) [inline articles/v14-23-upgrade/essential-steps/instructions/_common.adoc:172]: nit: this is already mentioned [here](https://vaadin.com/docs/latest/v14-23-upgrade/essential-steps/#replace-api-usage)

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc [hunk 2]
```diff
@@ -134,21 +134,49 @@ A set of API breaking changes and their replacements are listed below:
...
 To fix this, add the `router-ignore` attribute as in the following example:
-
++
 [source, java]
 ----
 Anchor logoutLink = new Anchor("/logout", "Logout from app");
 logoutLink.getElement().setAttribute("router-ignore", true);
 ----
 
+- Opening a Dialog when a modal Dialog is already open automatically makes the new Dialog a child of the modal Dialog.
```
Comment (joheriks) [inline articles/v14-23-upgrade/essential-steps/instructions/_common.adoc:155]: Would it make sense to add a sentence of context here in the beginning of this paragraph?. E.g., "Vaadin 23 introduces some changes in how modal components behave" (could later add a link to the page introduced in [this PR](https://github.com/vaadin/docs/pull/1236) further describing the server-side modality concept).

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc [hunk 3]
```diff
@@ -134,21 +134,49 @@ A set of API breaking changes and their replacements are listed below:
...
+Closing the modal also closes all its children.
```
Comment (joheriks) [inline articles/v14-23-upgrade/essential-steps/instructions/_common.adoc:156]:
```suggestion
Closing the modal Dialog also closes all its children.

```

**Review by joheriks — CHANGES_REQUESTED** [review]

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc (hunk 1 above)
Comment (tarekoraby) [reply articles/v14-23-upgrade/essential-steps/instructions/_common.adoc:172]: I think it was removed from the "Replace API Usage" section and moved here.

**Review by tarekoraby — APPROVED** [review]

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc [hunk 4]
```diff
@@ -134,21 +134,49 @@ A set of API breaking changes and their replacements are listed below:
...
+secondDialog.open();
+----
+
+-  Positive tabindex values (such as `setTabIndex(5)`) are no longer supported on input fields. Doing this will not cause errors, but will simply not have any effect on tab order.
+Tabindex values of `0` (to add a tab stop) and `-1` (to remove a tab stop) are still supported however.
+It is recommended to ensure that input fields are in the correct order in the DOM, instead of overriding the tab order with `setTabIndex`.
+
+- The DOM structure of many components has changed since V14. While this has no effect on their regular Java APIs, code utilizing Flow's Element API to access a component's internal elements may be affected.
+
+
```
Comment (tarekoraby) [inline articles/v14-23-upgrade/essential-steps/instructions/_common.adoc:178]:
```suggestion

- For Vaadin Charts, if [classname]`XAxis` or [classname]`YAxis` are added to a chart, then the ticks of those axes are now hidden by default.
To display the ticks, provide the axes with a tick width greater than 0 using [methodname]`XAxis.setTickWidth()` and [methodname]`YAxis.setTickWidth()`.

```

**Review by tarekoraby — COMMENTED** [review]
This would close #1178. But I can do it in a separate PR as well.

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc (hunk 1 above)
Comment (rolfsmeds) [reply articles/v14-23-upgrade/essential-steps/instructions/_common.adoc:172]: yup

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc (hunk 2 above)
Comment (rolfsmeds) [reply articles/v14-23-upgrade/essential-steps/instructions/_common.adoc]: Yes. Rewrote this bullet entirely in a new commit.

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc (hunk 1 above)
Comment (tomivirkki) [reply articles/v14-23-upgrade/essential-steps/instructions/_common.adoc:175]: Ah, yes, so it seems. The wording was also changed so I didn't realize it was a move. Nevermind.

File: articles/v14-23-upgrade/essential-steps/instructions/_common.adoc [hunk 5]
```diff
@@ -134,21 +134,55 @@ A set of API breaking changes and their replacements are listed below:
...
 Anchor logoutLink = new Anchor("/logout", "Logout from app");
 logoutLink.getElement().setAttribute("router-ignore", true);
 ----
 
+- The server-side modality mechanism introduced in V23 changes the behavior of modal Dialogs.
+Opening another Dialog when a modal Dialog is already open automatically makes the new Dialog a child of the previously opened one.
+Closing the "parent" modal also closes all subsequently opened Dialogs.
+
+To keep other Dialogs open when closing a modal Dialog, add them to the UI explicitly before opening:
++
```
Comment (tarekoraby) [inline articles/v14-23-upgrade/essential-steps/instructions/_common.adoc:161]:
```suggestion


```

**Review by joheriks — APPROVED** [review]

**Pull request merged**